python/requirements_pinned.txt

```diff
@@ -0,0 +1,61 @@
+## setup.py install_requires
```

Comment:
what's the plan to update this over time? Is this file generated by CI? Should updating this be added as a step in the release process?

Reply:
Generally the file should be automatically tracked by dependabot (we may have to rename it for that, I'll check that shortly). But yes, I'd suggest adding this to the release process